src/webserver/SessionActions.jl

```diff
@@ -18,9 +18,22 @@ function Base.showerror(io::IO, e::UserError)
     print(io, e.msg)
 end
 
+function change_notebook_id(session::ServerSession, nb::Notebook, new_id::UUID)
+    haskey(session.notebooks, nb.notebook_id) && delete!(session.notebooks, nb.notebook_id)
+    session.notebooks[new_id] = nb
+    nb.notebook_id = new_id
+    nb
+end
+
 function open_url(session::ServerSession, url::AbstractString; kwargs...)
```

Comment:
Talked about this offline, we will add a `notebook_id` kwarg to all the SessionAction functions so that this notebook id change is not needed